### PR TITLE
pythonPackages.pygal: fix build by forcing pytest_3

### PR DESCRIPTION
**Pull request by risicle**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by risicle** [comment]
cc @dotlambda

**Comment by dotlambda** [comment]
Alternatively, we can use `pytest_3`. But we'll also have to overhaul `WebOb`'s dependencies to prevent multiple versions of pytest in the closure.

Furthermore, would you mind getting rid of `buildInputs` in favor of `nativeBuildInputs` and `checkInputs`?

**Comment by risicle** [comment]
I don't really see the connection to `WebOb`, but I can see that `WebOb` appears to have `pytest` in its `propagatedBuildInputs`, which seems wrong.

**Comment by dotlambda** [comment]
> I don't really see the connection to `WebOb`

If you use `pytest_3` for `pygal`, you'll get
```
Found duplicated packages in closure for dependency 'pytest':
  pytest 3.10.1 (/nix/store/b7z68j94dfmfcyl371pqr6ggdsrggwk0-python3.7-pytest-3.10.1/lib/python3.7/site-packages)
  pytest 4.2.1 (/nix/store/jky420fbml8s6b563zzfci3wicni0f5b-python3.7-pytest-4.2.1/lib/python3.7/site-packages)
```
which is due to `WebOb`.

**Comment by risicle** [comment]
Not if I can help it...

**Comment by risicle** [comment]
@dotlambda ok, I've fixed up WebOb & routes to accommodate the switch of `pygal` to `pytest_3`. I had to create a `pytestrunner_pytest_3` to go with it though.

There are still downstream rebuilds that don't build, but only those that were already broken.

Noting that there are so many rebuilds, should this be targeted at `staging` instead of `master`?

**Comment by risicle** [comment]
Added a `staging` rebase of this https://github.com/NixOS/nixpkgs/pull/60888

File: pkgs/development/python-modules/pygal/default.nix [hunk 1]
```diff
@@ -29,11 +29,11 @@ buildPythonPackage rec {
 
     # Should be a check input, but upstream lists it under "setup_requires".
     # https://github.com/Kozea/pygal/issues/430
-    pytestrunner
+    (pytestrunner.overrideAttrs (oldAttrs: oldAttrs // { pytest = pytest_3; }))
```
Comment (FRidh) [inline pkgs/development/python-modules/pygal/default.nix]: pytestrunner.override

Comment (risicle) [reply pkgs/development/python-modules/pygal/default.nix]: Yeah I should probably close this PR if I'm not going to update it and focus on the `-staging` rebase instead https://github.com/NixOS/nixpkgs/pull/60888

**Comment by FRidh** [comment]
No need to open new PR's when rebasing.

**Comment by risicle** [comment]
It's not so much the rebase that caused it, it's that I was re-pointing it to `staging` rather than `master`. If there's a way to re-target a PR, I'd love to know.

**Comment by FRidh** [comment]
top right next to the PR title press edit, a branch selector pops up.

**Comment by risicle** [comment]
:astonished: you've changed my life.

**Pull request closed without merging**